src/components/ContrastChecker/GenerateContrastGridColors.jsx

```diff
@@ -86,13 +87,14 @@ const GenerateContrastGridColors = ({ baseColor, colorName = "primary" }) => {
 
           return (
             // Individual shade block
-            <div
+            <button
               key={shade.hex()} // Unique key for React rendering.
               className="p-2 text-center"
               style={{
                 backgroundColor: shade.hex(), // Set background to the shade color.
                 color: textColor, // Set text color for readability.
               }}
+              onClick={() => handleCopy(shade.hex().toUpperCase())} //Copy Hex value to clipboard
```

Comment:
You can create a `handleCopy` function in this file, and inside that we can call the `copyToClipboard` function, pass the data and call the toast inside `handleCopy`